### PR TITLE
Upgrading React to 17.X

### PR DESCRIPTION
**Pull request by ecaoile**
I rewrote the existing front end tests, but the main purpose of this PR is to make sure everything is working on the latest version of React.

### CODE REVIEW

File: client/src/__tests__/Header.test.js [hunk 1]
```diff
@@ -1,66 +1,39 @@
...
-    );
+    const eventsTag = screen.getByText('Event Calendar');
+    expect(eventsTag).toBeDefined();
 
-    // inspect the fourth anchor tag
-    const fourthATag = aTags.at(3);
-    expect(fourthATag.html()).toBe(
-      '<a class="nav-link ml-3 mr-3 text-center" href="/roles">Volunteer Roles</a>'
-    );
+    const vounteerTag = screen.getByText('Volunteer Roles');
```
Comment (jchi2241) [inline client/src/__tests__/Header.test.js]: `vounteer` typo

File: client/src/__tests__/Header.test.js [hunk 2]
```diff
@@ -1,66 +1,39 @@
 import React from 'react';
-import { mount } from 'enzyme';
+import '@testing-library/jest-dom';
+import { render, fireEvent, screen } from '@testing-library/react';
+
 import Header from '../components/Header';
```
Comment (jchi2241) [inline client/src/__tests__/Header.test.js]: nit. i believe you can reference stuff as absolute imports now, relative to the src dir.
```
import Header from 'components/Header';
```

**Review by jchi2241 — APPROVED** [review]
lgtm

File: client/src/__tests__/Header.test.js (hunk 2 above)
Comment (ecaoile) [reply client/src/__tests__/Header.test.js]: Oops, that was legacy import from before you made that change. Will modify

**Pull request merged**